src/GraphQl/Serializer/ItemNormalizer.php

```diff
@@ -110,6 +111,14 @@ public function normalize(mixed $object, string $format = null, array $context =
      */
     protected function normalizeCollectionOfRelations(ApiProperty $propertyMetadata, iterable $attributeValue, string $resourceClass, ?string $format, array $context): array
     {
+        if ('item_query' === $context['graphql_operation_name'] && $this->resourceMetadataCollectionFactory) {
```

Comment:
not a huge fan of this, can we use `$operation instanceof Query && !$operation instanceof CollectionOperationInterface`

Reply:
also put `$this->resourceMetadataCollectionFactory` first

Reply:
You are right, this is not pretty nice here. We do not have the `operation` yet, it is not in the `$context` - only the 'graphql_operation_name' is already there. Do not know exactly how to change that. Could you assist?

Reply:
Refactored and now works without this ugly check